### DISCUSSION TITLE
Abbr support in Markdown

### DISCUSSION THREAD

**Discussion by jcubic** [11ty/eleventy#1900] [Enhancement Queue]
In Jekyll, you can use abbreviations in markdown code like:

```
CSS is nice.

*[CSS]: Cascading Style Sheets
```

and CSS will be replaced with `<abbr title="Cascading Style Sheets">CSS</abbr>` (the last line will be removed from output) It would be nice if eleventy supports this as well. Or maybe you know any way to add this feature using plugins? I'm pretty new to Eleventy.

**Comment by pdehaan** [comment]
You might need to use https://github.com/markdown-it/markdown-it-abbr and then set a custom Markdown processor lib.

But it might be trickier if you want that to work in Nunjucks or LiquidJS templates.

**Reply by jcubic** [reply]
Thanks. Will check this out.

**Reply by pdehaan** [reply]
This might help you get started (or help a future traveller).

```js
// .eleventy.js
const markdownIt = require("markdown-it");

module.exports = function (eleventyConfig) {
  const mdOpts = {
    html: true,
    breaks: true,
    linkify: true,
  };
  const mdLib = markdownIt(mdOpts)
    .use(require("markdown-it-abbr"));
  eleventyConfig.setLibrary("md", mdLib);

  return {
    dir: {
      input: "src",
      output: "www",
    },
  };
};
```

```liquid
---
# ./src/index.md
title: 11ty-markdown-it-abbr test
layout: layouts/base
---

Some HTML and CSS are best. JS is rarely abbreviated.

{% include abbr %}
```

And my `./src/_includes/abbr.liquid file looks like this:

```liquid
*[HTML]: Hyper Text Markup Language
*[CSS]:  Cascading style sheets
*[JS]: JavaScript
```

But by including the abbreviations as a partial, it would let you share them between templates potentially. One thing that didn't work when I tried, was specifying the abbreviations in a shared layout file. I guess the markdown processing happens before handing it off to a layout file or some parent template. 🤷

**Comment by stevenvachon** [comment]
Here's a minor update to the code above to amend the existing markdown-it instance instead of creating a new one:
```js
import MarkdownItAbbr from 'markdown-it-abbr';

export default config => {
  config.amendLibrary('md', (mdLib) => mdLib.use(MarkdownItAbbr));

  return {
    dir: {
      // etc
    },
  };
};
```

**Reply by Ryuno-Ki** [reply]
TIL! Thank you.
